app/src/main/java/com/rkm/tasky/database/model/EventWithDetails.kt

```diff
@@ -0,0 +1,21 @@
+package com.rkm.tasky.database.model
+
+import androidx.room.Embedded
+import androidx.room.Relation
+
+data class EventWithDetails(
+    @Embedded val event: EventEntity,
+
+    @Relation(
+        parentColumn = "id",
+        entityColumn = "eventId"
+    )
+    val attendees: List<AttendeeEntity>,
+
+    @Relation(
+        parentColumn = "id",
+        entityColumn = "eventId"
+    )
+    val photos: List<PhotoEntity>
+
+)
```

Comment:
This will work fine and I initially also has the attendees in a separate table, but I found that there was no purpose to have them in that separate table as I never needed them on their own, I always accessed them via an event object. So I simplified my approach to include the attendees with the entity model, check it out:

```
@Entity
data class EventEntity(
    @PrimaryKey(autoGenerate = false)
    val id: String,
    val title: String,
    val description: String?,
    val hostId: String,
    val isUserEventCreator: Boolean,
    val fromTime: Long,
    val toTime: Long,
    val remindBefore: RemindBeforeDuration,
    val photos: List<PhotoSerialized>,
    val attendees: List<AttendeeSerialized>,
)
```

Here is the serialized models (they have domain counterparts):
```
@Serializable
data class AttendeeSerialized(
    val userId: String,
    val email: String,
    val name: String,
    val eventId: String,
    val isGoing: Boolean,
    val remindAt: Long,
    val isCreator: Boolean,
)

@Serializable
data class PhotoSerialized(
    val key: String,
    val url: String
)
```

And here is how I map between the domain and entity models:
```
fun AgendaItem.toEventEntity(): EventEntity? {
    return details.detailsAsEvent()?.let { eventDetails ->
        EventEntity(
            id = id,
            title = title,
            description = description,
            hostId = eventDetails.hostId,
            isUserEventCreator = eventDetails.isUserEventCreator,
            fromTime = time.toMillis(),
            toTime = eventDetails.toTime.toMillis(),
            remindBefore = remindBeforeDuration,
            photos = eventDetails.photos.toSerializedPhotos(),
            attendees = eventDetails.attendees.toSerializedAttendees()
        )
    }
}

fun EventEntity.toAgendaItem(): AgendaItem {
    return AgendaItem(
        id = id,
        title = title,
        description = description ?: "",
        remindBeforeDuration = remindBefore,
        time = fromTime.toLocalDateTime(),
        details = AgendaItemDetails.Event(
            toTime = toTime.toLocalDateTime(),
            attendees = attendees.toAttendees(),
            photos = photos.toPhotos(),
            hostId = hostId,
            isUserEventCreator = isUserEventCreator,
        )
    )
}
```